### PR TITLE
[API-5904] - Deprecate ClaimsApi "schema" and "validate" endpoints

### PR DESCRIPTION
**Pull request by jw81**
## Description of change

Deprecates the "schema" and "validate" endpoints for both v0 and v1 of ClaimsApi.
- Updates Swagger to visually indicate that the endpoints are deprecated.
- Adds logic to return `Deprecation` and optional `Link` response headers

## Original issue(s)
[API-5904](https://vajira.max.gov/browse/API-5904)

### CODE REVIEW

File: modules/claims_api/app/controllers/concerns/claims_api/endpoint_deprecation.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  module EndpointDeprecation
+    extend ActiveSupport::Concern
+
+    included do
+      V0_DEV_DOCS = 'https://developer.va.gov/explore/benefits/docs/claims?version=0.0.1'
+      V1_DEV_DOCS = 'https://developer.va.gov/explore/benefits/docs/claims?version=current'
```
Comment (SlakrHakr) [inline modules/claims_api/app/controllers/concerns/claims_api/endpoint_deprecation.rb:9]: might be worth putting these in Settings instead?

Comment (SlakrHakr) [inline modules/claims_api/app/controllers/concerns/claims_api/endpoint_deprecation.rb:9]: i wonder if its worth using
https://developer.va.gov/explore/benefits/docs/claims?version=1.0.0
as the url instead.

the above url still redirects to "current" it looks like, but should stay v1 if a v2 is introduced as the new "current" version.

Comment (jw81) [reply modules/claims_api/app/controllers/concerns/claims_api/endpoint_deprecation.rb]: Ah, ok.
Yeah, I had originally put `version=1.0.0` but when it redirected to "current", I wasn't sure if maybe that url would "break" whenever a new version comes out.
Updated [here](https://github.com/department-of-veterans-affairs/vets-api/pull/6289/commits/8aa37123798926ef1d594dbe486052c0d97392d5).

Comment (jw81) [reply modules/claims_api/app/controllers/concerns/claims_api/endpoint_deprecation.rb]: Do you mean storing them [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/config/settings.yml#L414)?

**Review by thilton-oddball — APPROVED** [review]
Looks good!

File: modules/claims_api/app/controllers/concerns/claims_api/endpoint_deprecation.rb (hunk 1 above)
Comment (SlakrHakr) [reply modules/claims_api/app/controllers/concerns/claims_api/endpoint_deprecation.rb]: yes

**Review by SlakrHakr — APPROVED** [review]

**Pull request merged**